### DISCUSSION TITLE
Enhanced Wire Navigation with Diffing and Page Caching Options

### DISCUSSION THREAD

**Discussion by joydeep-bhowmik** [livewire/livewire#6136] [3. Feature ideas]
Is it possible to implement wire navigation with diffing similar to React or Vue, rather than replacing the entire body element? Additionally, can we cache each visited page automatically upon push state and refresh it when data changes, or selectively choose not to cache specific pages?

**Comment by calebporzio** [comment]
Hmmm yeah both good questions. To keep things simple and get this part of Livewire stable I'm going to say no to both of those. But after wire:navigate is battle-tested and stable I'd like to add further optimizations like those. Thanks for askin
